libmscore/score.cpp

```diff
@@ -2987,14 +2987,13 @@ void Score::deselect(Element* el)
 
 void Score::select(Element* e, SelectType type, int staffIdx)
       {
-      if (e && (e->isNote() || e->isRest())) {
-            Element* ee = e;
-            if (ee->isNote())
-                  ee = ee->parent();
-            Fraction tick = toChordRest(ee)->segment()->tick();
-            if (masterScore()->playPos() != tick)
-                  masterScore()->setPlayPos(tick);
+      // Move the playhead to the selected element's preferred play position.
+      if (e) {
+            const auto playTick = e->playTick();
+            if (masterScore()->playPos() != playTick)
+                  masterScore()->setPlayPos(playTick);
             }
```

Comment:
unexpectedly see the playTick method for elements such as BarLine, InstrumentName, etc. It is better to make a method, or even a separate class, that will determine the playTick for the selected element.
Something like:
```
if (e) {
            const Fraction playTick = PlayTickResolver::playTickForSelectedElement(e);
            if (masterScore()->playPos() != playTick)
                  masterScore()->setPlayPos(playTick);
            }
```

Reply:
The `playTick()` function is analogous to, and sits nicely beside, the `tick()` and `rtick()` virtual functions that already exist in the `Element` class hierarchy. Allowing each element type to determine its own play tick position achieves much better encapsulation than doing it from the outside.

Having a separate helper class that would return a variety of different results after manually checking against a curated laundry list of element types would sort of defeat the purpose of having the class hierarchy in the first place. (In that case, we might have a `TickResolver` helper class, an `RTickResolver` helper class, and so on.)

FWIW, my first draft implementation of this feature actually worked exactly that way: It contained a convoluted `switch` statement far removed from the element subclasses that it was governing. It wasn't pretty.